Amazon.IonObjectMapper/Serializer/IonObjectSerializer.cs

```diff
@@ -309,163 +377,118 @@ private bool IsBackingFieldForReadonlyProperty(FieldInfo field)
         private IEnumerable<(MethodInfo, string)> GetGetters()
         {
             var getters = new List<(MethodInfo, string)>();
-            foreach (var method in targetType.GetMethods(BINDINGS))
+            foreach (var method in this.targetType.GetMethods(BINDINGS))
             {
-                var getMethod = (IonPropertyGetter)method.GetCustomAttribute(typeof(IonPropertyGetter));
-                
+                var getMethod = (IonPropertyGetterAttribute)method.GetCustomAttribute(typeof(IonPropertyGetterAttribute));
+
                 // A getter method should have zero parameters.
                 if (getMethod?.IonPropertyName == null || method.GetParameters().Length != 0)
                 {
                     continue;
                 }
-                
+
                 getters.Add((method, getMethod.IonPropertyName));
             }
 
             return getters;
         }
-        
+
         private MethodInfo FindSetter(string name)
         {
-            return targetType.GetMethods(BINDINGS).FirstOrDefault(m =>
+            return this.targetType.GetMethods(BINDINGS).FirstOrDefault(m =>
             {
-                var setMethod = (IonPropertySetter)m.GetCustomAttribute(typeof(IonPropertySetter));
+                var setMethod = (IonPropertySetterAttribute)m.GetCustomAttribute(typeof(IonPropertySetterAttribute));
                 return setMethod != null && setMethod.IonPropertyName == name;
             });
         }
 
         private string IonFieldNameFromProperty(PropertyInfo property)
         {
-            var ionPropertyName = property.GetCustomAttribute(typeof(IonPropertyName));
-            if (ionPropertyName != null) 
+            var ionPropertyName = property.GetCustomAttribute(typeof(IonPropertyNameAttribute));
+            if (ionPropertyName != null)
             {
-                return ((IonPropertyName)ionPropertyName).Name;
+                return ((IonPropertyNameAttribute)ionPropertyName).Name;
             }
 
-            return options.NamingConvention.FromProperty(property.Name);
+            return this.options.NamingConvention.FromProperty(property.Name);
         }
 
         private PropertyInfo FindProperty(string readName)
         {
-            var exact = GetValidProperties(false).Where(IsIonNamedProperty).FirstOrDefault(p => 
+            var exact = this.GetValidProperties(false).Where(IsIonNamedProperty).FirstOrDefault(p =>
                 {
-                    var ionPropertyName = p.GetCustomAttribute<IonPropertyName>();
+                    var ionPropertyName = p.GetCustomAttribute<IonPropertyNameAttribute>();
                     if (ionPropertyName != null)
                     {
-                        return p.GetCustomAttribute<IonPropertyName>().Name == readName;
+                        return p.GetCustomAttribute<IonPropertyNameAttribute>().Name == readName;
```

Comment:
Why aren't we using `ionPropertyName` here?

Reply:
What do you mean? I'm unsure since I didn't write this code, but this is in a `Where` lambda where it expects a boolean, which this is.

Reply:
Nvm I see what you meant